### PR TITLE
Move docker release to docker folder and legacy script for tar to scripts folder

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
Move docker release to docker folder and legacy script for tar to scripts folder.
Please only merge this when #710 merged.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docker/release/README.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+* Scenario 3: No demo certs/configs + enable security on both OpenSearch and OpenSearch-Dashboards (cluster exit with errors due to demo install script is not run. Therefore, no certs/configs are available for Security Plugin. Useful if you want to setup your own certs/configs):
+  * OpenSearch:
+     ```
+     $ docker run -it -p 9200:9200 -p 9600:9600 -e "discovery.type=single-node" -e "DISABLE_INSTALL_DEMO_CONFIG=true" opensearchproject/opensearch:1.1.0
+     ```
+  * Dashboards:
+     ```
+     $ docker run -it --network="host" -e opensearchproject/opensearch-dashboards:1.1.0
+     ```
+
```
Comment (dblock) [inline docker/release/README.md:107]: A general comment, bullets are really hard to follow, make sections like `#### Scenario 3`.

File: docker/release/build-image-multi-arch.sh [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        echo -e "\nCopying ${PRODUCT} arch ${ARCHITECTURE_ARRAY[$index]} version ${VERSION} from ${TARBALL_ARRAY[$index]}"
+        cp -v ${TARBALL_ARRAY[$index]} ${DIR}/${PRODUCT}-${arch_uname}.tgz
+    done
+    ls -l $DIR
+fi
+
+# Build multi-arch images
+PLATFORMS=`echo "${ARCHITECTURE_ARRAY[@]/#/linux/}" | sed 's/x64/amd64/g;s/ /,/g'` && echo PLATFORMS $PLATFORMS
+docker buildx build --platform $PLATFORMS --build-arg VERSION=$VERSION --build-arg BUILD_DATE=`date -u +%Y-%m-%dT%H:%M:%SZ` -t ${REPOSITORY}:${VERSION} -t ${REPOSITORY}:latest -f $DOCKERFILE --push $DIR
+
```
Comment (dblock) [inline docker/release/build-image-multi-arch.sh:155]: We should consider writing these in Python, it's a lot less error prone and would be 5-10 lines of code.

The first 100 lines become 5 lines of `argparse`.
The next 100 lines become 3 classes.
It would also be testable.

File: docker/release/README.md (hunk 1 above)
Comment (peterzhuamazon) [reply docker/release/README.md:107]: Got it. Will improve.

Comment (peterzhuamazon) [reply docker/release/README.md:105]: improved.

File: docker/release/README.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
+## OpenSearch/Dashboards Docker Build README
```
Comment (dblock) [inline docker/release/README.md:1]: This is a README, remove from title

File: docker/release/README.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+### Summary
+We support building OpenSearch/Dashboards docker images with [multi-CPU architecture support](https://docs.docker.com/desktop/multi-arch/).
```
Comment (dblock) [inline docker/release/README.md:4]: "support" used twice

File: docker/release/README.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+We also support old school single CPU architecture build as well.
```
Comment (dblock) [inline docker/release/README.md:5]: "also" and "as well" -> ... replace this whole thing with 

```
We support building OpenSearch and OpenSearch Dashboards docker [multi-CPU](https://docs.docker.com/desktop/multi-arch/) and single CPU architectures.
```

File: docker/release/README.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+[OpenSearch-Dashboards Docker Repository](https://hub.docker.com/r/opensearchproject/opensearch-dashboards/)
+
+```
+docker pull opensearchproject/opensearch:latest
+docker pull opensearchproject/opensearch-dashboards:latest
+```
+
+### Building Docker Images
+#### We provide two scripts to build docker images. For single-arch image you need to install just the Docker Engine on your host machine. For multi-arch image (currently support x64/arm64) you need to install Docker Desktop (macOS/Windows) or Docker Buildx (LINUX).
```
Comment (dblock) [inline docker/release/README.md:18]: Why is this a section title?

File: docker/release/README.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+* Install Docker through the official docker webpage: https://docs.docker.com/get-docker/
```
Comment (dblock) [inline docker/release/README.md:20]: I would just remove bullets.

File: docker/release/README.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+* Verify if you have Docker Desktop or Docker Buildx Standalone by running:
+  ```
+  docker buildx --help
+  ```
+
+* You need to run both script within the `opensearch-build/release/docker` folder. Running them
+  within other path would cause the scripts to fail.
+
+#### Build single-arch image with these command:
```
Comment (dblock) [inline docker/release/README.md:35]: commands, missing an s

File: docker/release/README.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+docker pull opensearchproject/opensearch:latest
+docker pull opensearchproject/opensearch-dashboards:latest
+```
+
+### Building Docker Images
+#### We provide two scripts to build docker images. For single-arch image you need to install just the Docker Engine on your host machine. For multi-arch image (currently support x64/arm64) you need to install Docker Desktop (macOS/Windows) or Docker Buildx (LINUX).
+
+* Install Docker through the official docker webpage: https://docs.docker.com/get-docker/
+
+* After installation, verify whether or not you have Docker Engine by running:
```
Comment (dblock) [inline docker/release/README.md:22]: verify whether or not -> verify

**Review by dblock — APPROVED** [review]
I have a bunch of nitpicky comments on the README, that can generally be shortened and lots of words removed. Feel free to address or not.

File: docker/release/README.md (hunk 3 above)
Comment (peterzhuamazon) [reply docker/release/README.md:1]: fixed.

File: docker/release/README.md (hunk 4 above)
Comment (peterzhuamazon) [reply docker/release/README.md:4]: fixed.

File: docker/release/README.md (hunk 5 above)
Comment (peterzhuamazon) [reply docker/release/README.md:5]: thanks.

File: docker/release/README.md (hunk 6 above)
Comment (peterzhuamazon) [reply docker/release/README.md:18]: This is needed as it separate build image, configure run, and actually running the docker container.

File: docker/release/README.md (hunk 7 above)
Comment (peterzhuamazon) [reply docker/release/README.md:20]: Removed.

File: docker/release/README.md (hunk 9 above)
Comment (peterzhuamazon) [reply docker/release/README.md:22]: Thanks.

**Review by peternied — APPROVED** [review]
